src/main/java/vimification/model/task/Task.java

```diff
@@ -3,51 +3,48 @@
 import static java.util.Objects.requireNonNull;
 import static vimification.commons.util.CollectionUtil.requireAllNonNull;
 
-import vimification.model.task.components.Description;
-
 public abstract class Task {
 
-    private final Description description;
-    private final Status status;
+    private String description;
+    private boolean isDone;
     private final TaskType type;
 
     /**
      * Every field must be present and not null.
      */
-    public Task(Description description, Status status, TaskType type) {
-        requireAllNonNull(description, status);
+    public Task(String description, boolean isDone, TaskType type) {
```

Comment:
Probably change this to `protected` - or just use the no modifier (package private). This will not be used outside of the package.